### PR TITLE
provide a better error message for missing json libraries

### PR DESCRIPTION
**Pull request by Tieske**

### CODE REVIEW

File: src/luarocks/upload/api.lua [hunk 1]
```diff
@@ -126,7 +135,7 @@ if not ltn12_ok then -- If not using LuaSocket and/or LuaSec...
 function Api:request(url, params, post_params)
    local vars = cfg.variables
    local json_ok, json = require_json()
-   if not json_ok then return nil, "A JSON library is required for this command." end
+   if not json_ok then return nil, "A JSON library is required for this command. "..json end
```
Comment (daurnimator) [inline src/luarocks/upload/api.lua]: this same log message is in multiple places.

I'll also note that this is a minor merge conflict with #466

**Comment by daurnimator** [comment]
Btw, these json libraries have slightly different apis e.g. on invalid json:
  - cjson throws an error 
  - dkjson returns `nil, position, error_message`

Is this something we need to handle?

File: src/luarocks/upload/api.lua (hunk 1 above)
Comment (Tieske) [inline src/luarocks/upload/api.lua]: Thx. good catch. I fixed the 2nd occurence.

**Comment by Tieske** [comment]
> Btw, these json libraries have slightly different apis e.g. on invalid json:

Not until it proves a problem imo

**Comment by mpeterv** [comment]
LGTM

**Pull request merged**